keylime/cloud_verifier_tornado.py

```diff
@@ -433,11 +443,58 @@ def post(self):
                             logger.warning(e.message)
                             return
                     else:
-                        # Read legacy allowlist if present in payload
                         ima_policy = json_body.get("allowlist", None)
-                        if not ima_policy:
-                            logger.info("Allowlist data not provided with request! Using default allowlist.")
-                            ima_policy = json.dumps(ima.process_ima_policy(ima.EMPTY_ALLOWLIST, []))
+
+                    if ima_policy_name:
+                        # Find whether an allowlist with the provided name already exists
+                        try:
+                            ima_policy_stored = session.query(
+                                VerifierAllowlist).filter_by(name=ima_policy_name).one_or_none()
+                            logger.info("ima_policy_stored is: %s", ima_policy_stored)
+                        except SQLAlchemyError as e:
+                            logger.error('SQLAlchemy Error: %s', e)
+                            raise
+
+                        # Don't allow owerwriting existing IMA policies
+                        if ima_policy and ima_policy_stored:
+                            web_util.echo_json_response(
+                                self, 409, f"Allowlist with name {ima_policy_name} already exists")
+                            logger.warning(
+                                "Allowlist with name %s already exists", ima_policy_name)
+                            return
+
+                    if not ima_policy_name and not ima_policy:
+                        logger.info("Allowlist data not provided with request! Using default allowlist.")
```

Comment:
Maybe we can take this opportunity to improve this message to let the user know the default allowlist is empty. Maybe something like "Allowlist data not provided with request! Using default empty allowlist"

Reply:
Done!